geode-core/src/main/java/org/apache/geode/internal/net/SocketUtils.java

```diff
@@ -89,6 +89,10 @@ public static boolean close(final ServerSocket serverSocket) {
    * and buffer.remaining is also zero the limit is changed to be buffer.capacity
    * before reading.
    *
+   * @param socket the socket to read from
+   * @param inputBuffer the buffer into which data should be stored
+   * @param socketInputStream the socket's inputStream, included as a parameter so it can be a
+   *        buffered stream, if desired
```

Comment:
"if desired" by whom—by what part of the code? The comment implies the caller controls whether or not the stream is used, perhaps by making it `null` or not. Reality is this parameter is dependent on the kind of object referenced by the `socket` parameter.

Any time I see a dependency between parameters like this, especially when it comes with a switch-on-type it makes me wonder if there is a mis-assignment of responsibility somewhere. Is there already some object that could hold the stream? If so, could we delegate this behavior to that object?

Reply:
The older Old IO code base used a BufferedInputStream wrapper around the socket's input stream on recommendation from StackOverflow and other sources.  I did not find that it increased performance so I removed that wrapper, but I left this parameter in place for flexibility since this is a "Util" class.  This parameter also makes the methods easier to test.

Reply:
thanks